consensus/src/quorum_store/batch_proof_queue.rs

```diff
@@ -715,11 +715,11 @@ impl BatchProofQueue {
     }
 
     pub(crate) fn handle_updated_block_timestamp(&mut self, block_timestamp: u64) {
+        // tolerate asynchronous notification
+        if self.latest_block_timestamp > block_timestamp {
+            return;
+        }
         let start = Instant::now();
-        assert!(
-            self.latest_block_timestamp <= block_timestamp,
-            "Decreasing block timestamp"
-        );
         self.latest_block_timestamp = block_timestamp;
```

Comment:
Should we only update when `self.latest_block_timestamp <= block_timestamp`?

Reply:
the early return above does it?